analyzers/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -87,12 +87,20 @@ public abstract class UtilityAnalyzerBase<TMessage> : UtilityAnalyzerBase
             context.RegisterCompilationAction(c =>
                 {
                     ReadParameters(context, c);
-                    if (!IsAnalyzerEnabled ||
-                        (SkipAnalysisForTestProject && IsTestProject))
+                    if (!IsAnalyzerEnabled)
                     {
                         return;
                     }
 
+                    // The results of Metrics and CopyPasteToken analyzers are not needed for Test projects yet the plugin side expects the protobuf files, so we create empty ones.
+                    if (SkipAnalysisForTestProject && IsTestProject)
+                    {
+                        // Make sure the folder exists
+                        Directory.CreateDirectory(OutPath);
+                        File.WriteAllBytes(Path.Combine(OutPath, FileName), new byte[] { });
```

Comment:
I don't like the bytes there, as it raises the questions what does it mean to write something empty. Let's just create the file.
```suggestion
                        File.Create(Path.Combine(OutPath, FileName)).Dispose();
```